# Group 13 Activity Week 6
# Reading the countrywise GDP data and analysing it


In [49]:
import numpy as np
import pandas as pd

# Reading Data set 

In [50]:
dfGDP = pd.read_excel('Dataset_w6.xlsx')

In [51]:
dfGDP.head(5)

country          Region  Surface area (km2)  \
0          Afghanistan    SouthernAsia            652864.0   
1              Albania  SouthernEurope             28748.0   
2              Algeria  NorthernAfrica           2381741.0   
3               Angola    MiddleAfrica           1246700.0   
4  Antigua and Barbuda       Caribbean               442.0   

   Population in thousands (2017)  GDP per capita (current US$)  \
0                         35530.0                         623.2   
1                          2930.0                        3984.2   
2                         41318.0                        4154.1   
3                         29784.0                        4714.1   
4                           102.0                       14764.5   

   Agricultural production index   International trade: Imports (million US$)  
0                           125.0                                      3568.0  
1                           134.0                                      4669.0  
2                           160.0                                     47091.0  
3                           175.0                                      8790.0  
4                            88.0                                       491.0

# 1. Finding the number of countries in the dataset

In [52]:
print("There are " ,dfGDP.country.count() , " countries in the Dataset")

There are  196  countries in the Dataset


# Number of countries in each region

In [71]:
# creating a dataframe grouped by Region
dfGDP.groupby(by="Region").size() 

Region
Caribbean            16
CentralAmerica        8
CentralAsia           5
EasternAfrica        17
EasternAsia           7
EasternEurope        10
Melanesia             5
Micronesia            4
MiddleAfrica          9
NorthernAfrica        5
NorthernAmerica       4
NorthernEurope       10
Oceania               2
Polynesia             5
South-easternAsia    11
SouthAmerica         12
SouthernAfrica        5
SouthernAsia          9
SouthernEurope       12
WesternAfrica        15
WesternAsia          18
WesternEurope         7
dtype: int64

# 3 a.	On average which region has the highest GDP per capita (current US$
            


In [114]:
#Finding out the average GDP of each Region
df1 = dfGDP.groupby("Region")['GDP per capita (current US$)'].mean().round().reset_index()
df1

Region  GDP per capita (current US$)
0           Caribbean                      15135.56
1      CentralAmerica                       6348.14
2         CentralAsia                       4329.88
3       EasternAfrica                       2088.85
4         EasternAsia                      27967.61
5       EasternEurope                       9292.49
6           Melanesia                       9261.96
7          Micronesia                       5008.30
8        MiddleAfrica                       3928.83
9      NorthernAfrica                       3934.96
10    NorthernAmerica                      57659.00
11     NorthernEurope                      42098.22
12            Oceania                      44823.25
13          Polynesia                       8686.52
14  South-easternAsia                      10299.03
15       SouthAmerica                       8499.78
16     SouthernAfrica                       4191.90
17       SouthernAsia                       2968.40
18     SouthernEurope                      14462.74
19      WesternAfrica                       1013.14
20        WesternAsia                      17013.94
21      WesternEurope                      55387.13

In [136]:
#Finding out top 3  regions with highest averaged GDP
df1.sort_values('GDP per capita (current US$)').nlargest(3,'GDP per capita (current US$)')

Region  GDP per capita (current US$)
10  NorthernAmerica                      57659.00
21    WesternEurope                      55387.13
12          Oceania                      44823.25

# Region NorthernAmerica has the highest average GDP

# 3 b.	On average which region has the highest International trade: Imports (million US$)


In [137]:
df2 = dfGDP.groupby("Region")['International trade: Imports (million US$)'].mean().round(2).reset_index()
df2.sort_values('International trade: Imports (million US$)',ascending=False).nlargest(3,'International trade: Imports (million US$)')

Region  International trade: Imports (million US$)
10  NorthernAmerica                                   663552.25
4       EasternAsia                                   451995.00
21    WesternEurope                                   407498.43

#Region NorthernAmerica has the highest International trade: Imports (million US$)


# 4.	Find out which region has more consistent with respect to GDP per capita (current US$)

In [126]:
df3 = dfGDP.groupby("Region")['GDP per capita (current US$)'].std().round(2).reset_index()
df3.sort_values('GDP per capita (current US$)',ascending=False).nsmallest(3,'GDP per capita (current US$)')

Region  GDP per capita (current US$)
19   WesternAfrica                        815.10
9   NorthernAfrica                        974.71
16  SouthernAfrica                       2177.55

The Region WesternAfrica has the least standard deviation . So it has more consistent 'GDP per capita (current US$)

# Summary

There are 196 countries from 22 regions in the dataset given.
1 . Luxembourg in WesternEurope is the country with highest GDP of 100160.8 where as Somalia in East Africa is the least with     144.5 only.
2.13727.34 is the Average GDP of 196 countries
3.NorthernAmerica region has the highest average GDP where as WesternAfrica has the least
4.The Region WesternAfrica has the least standard deviation . So it has more consistent 'GDP per capita (current US$)
